x/stakeibc/keeper/community_pool.go

```diff
@@ -1,12 +1,104 @@
 package keeper
 
 import (
+	"errors"
+	"fmt"
+
+	sdkmath "cosmossdk.io/math"
 	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/Stride-Labs/stride/v14/utils"
 )
 
+// For each hostZone with a valid community pool, trigger the ICQs and ICAs to transfer deposited tokens to holding address
+// Since ICQs and ICAs take time to complete, it is almost certain tokens swept in and processed will be swept out in a later epoch
+func (k Keeper) SweepInAllDepositedCommunityPoolTokens(ctx sdk.Context) error {
+	hostZones := k.GetAllActiveHostZone(ctx)
+	for _, hostZone := range hostZones {
+		if hostZone.CommunityPoolDepositIcaAddress != "" &&
+			hostZone.CommunityPoolHoldingAddress != "" &&
+			hostZone.CommunityPoolReturnIcaAddress != "" {
```

Comment:
nit: We should add a helper function on the host zone type to validate these

```go
// types/host_zone.go

func (h HostZone) ValidateCommunityPoolAddresses() error {
   if h.CommunityPoolDepositIcaAddress == "" {
          return types.ErrICANotRegistered.Wrapf("community pool deposit ICA not registered")
   }
   ...
}
```

(out of scope point - we should also add this for our other ICA addresses at some point!)